Create function for exporting `dqdashboard_results` table to a json file

jshoughtaling opened the pull request:
Following up on a discussion with @katy-sadowski and in response to Issue #367:

This PR adds a function, `writeDBResultsToJson` that extracts data from the `dqdashboard_results` database table and exports it to a json file for use with other OHDSI tools like [AresIndexer](https://github.com/OHDSI/AresIndexer) or the shiny dashboard viewer.

The added functionality enhances the recently updated SQLOnly execution use case, where users can incrementally fill a database table with output of SQL checks executed with custom orchestration schemes.

Note that the exported json file may require a case change for use with AresIndexer depending on the version of that package.

katy-sadowski commented on R/writeDBResultsTo.R:
```diff
@@ -0,0 +1,104 @@
+#' @param connectionDetails         A connectionDetails object for connecting to the CDM database
+#' @param resultsDatabaseSchema     The fully qualified database name of the results schema
+#' @param cdmDatabaseSchema         The fully qualified database name of the CDM schema
+#' @param writeTableName            Name of table in the database to write results to
```
would update to "Name of DQD results table in the database to read from"

katy-sadowski commented on R/writeDBResultsTo.R:
```diff
@@ -0,0 +1,104 @@
+#' @param resultsDatabaseSchema     The fully qualified database name of the results schema
+#' @param cdmDatabaseSchema         The fully qualified database name of the CDM schema
+#' @param writeTableName            Name of table in the database to write results to
+#' @param outputFolder              The output folder
```
"The folder to output the json results file to"

katy-sadowski commented on R/writeDBResultsTo.R:
```diff
@@ -0,0 +1,104 @@
+                                    outputFile) {
+    startTime <- Sys.time()
+
+    sql <- SqlRender::render(
+          sql = "select * from @cdmDatabaseSchema.cdm_source;",
+          cdmDatabaseSchema = cdmDatabaseSchema
+        )
+
+    sql <- SqlRender::translate(
+         sql = sql,
+         targetDialect = connectionDetails$dbms
+        )
+
+    metadata <- DatabaseConnector::querySql(
+         connection = connection,
+         sql = sql,
+         snakeCaseToCamelCase = TRUE
+        )
+
```
i think you could simplify this with https://ohdsi.github.io/DatabaseConnector/reference/renderTranslateQuerySql.html

katy-sadowski replied:
ditto for the ones below

katy-sadowski commented on R/writeDBResultsTo.R:
```diff
@@ -0,0 +1,104 @@
+    # Quick patch for missing value issues related to SQL Only Implementation
+    checkResults["error"][checkResults["error"] == ''] <- NA
+    checkResults["warning"][checkResults["warning"] == ''] <- NA
+    checkResults["executionTime"][checkResults["executionTime"] == ''] <- '0.1 secs'
```
why 0.1?

katy-sadowski commented on R/writeDBResultsTo.R:
```diff
@@ -0,0 +1,104 @@
+    allResults <- list(
+        startTimestamp = Sys.time(),
+        endTimestamp = Sys.time(),
+        executionTime = sprintf("%.0f %s", delta, attr(delta, "units")),
```
maybe simpler just to put a placeholder value here as well, since this delta isn't super meaningful?

katy-sadowski commented on tests/testthat/test-writeDBResultsTo.R:
```diff
@@ -0,0 +1,42 @@
+
+  connection <- DatabaseConnector::connect(connectionDetailsEunomia)
+  tableNames <- DatabaseConnector::getTableNames(connection = connection, databaseSchema = resultsDatabaseSchemaEunomia)
+  expect_true("dqdashboard_results" %in% tolower(tableNames))
```
i think this can be removed since we already have another test for this

katy-sadowski commented on tests/testthat/test-writeDBResultsTo.R:
```diff
@@ -0,0 +1,42 @@
+    )
+
+  on.exit(DatabaseConnector::disconnect(connection), add = TRUE)
+  expect_true(file.exists(file.path(outputFolder,testExportFile)))
```
something else that could be good to test here - look at length of checkResults and compare to rowcount in the DB table

katy-sadowski reviewed (changes requested):
Suggested minor changes.

Also, want to update this doc to use your function? https://ohdsi.github.io/DataQualityDashboard/articles/SqlOnly.html

jshoughtaling commented:
> Suggested minor changes.
> 
> Also, want to update this doc to use your function? https://ohdsi.github.io/DataQualityDashboard/articles/SqlOnly.html

@katy-sadowski  - thanks so much for the quick review! 

I've implemented all changes as suggested, included the additional test of check rows, and modified the SQLOnly vignette to reference the new function prior to viewing json results.

katy-sadowski commented on R/writeDBResultsTo.R:
```diff
@@ -36,43 +36,27 @@ writeDBResultsToJson <- function(connection,
                                     outputFile) {
     startTime <- Sys.time()
```
can remove start and end time, and the delta variable now we're not it :)

jshoughtaling replied:
Done!

katy-sadowski commented on vignettes/SqlOnly.rmd:
```diff
@@ -171,49 +171,18 @@ for (dqdSqlFile in dqdSqlFiles) {
-)
+# Extract results table to JSON file for viewing or secondary use
+
+DataQualityDashboard::writeDBResultsToJson(
```
🤩  love to see those lines of code slashed!

katy-sadowski reviewed (changes requested):
1 more minor change
